File: scaffold/src/main/java/top/xuqingquan/web/publics/WebCreator.java
package top.xuqingquan.web.publics;

import android.widget.FrameLayout;
import top.xuqingquan.web.nokernel.IWebIndicator;

public interface WebCreator extends IWebIndicator {
    WebCreator create();

    android.webkit.WebView getWebView();

    com.tencent.smtt.sdk.WebView getX5WebView();

    FrameLayout getWebParentLayout();
}

File: scaffold/src/main/java/top/xuqingquan/web/publics/WebCreator.kt
package top.xuqingquan.web.publics

import android.widget.FrameLayout
import top.xuqingquan.web.nokernel.BaseIndicatorSpec
import top.xuqingquan.web.nokernel.IWebIndicator

interface WebCreator<T : BaseIndicatorSpec> : IWebIndicator<T> {

    fun getWebView(): android.webkit.WebView?

    fun getX5WebView(): com.tencent.smtt.sdk.WebView?

    fun getWebParentLayout(): FrameLayout

    fun create(): WebCreator<T>
}
